src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java

```diff
@@ -0,0 +1,146 @@
+				user.setCurrentHand(hand);
+				return TypedActionResult.success(stack);
+			}
+		}
+		return super.use(world, user, hand);
+	}
+
+	@Override
+	public ItemStack finishUsing(ItemStack stack, World world, LivingEntity user) {
+		if (world.isClient)
+			return stack;
+		ItemStack offStack = user.getStackInHand(Hand.OFF_HAND);
+		if (offStack.getItem() == AstromineItems.GRAVITY_GAUNTLET) {
+			EnergyHandler selfHandler = Energy.of(stack);
+			EnergyHandler otherHandler = Energy.of(offStack);
+			if (selfHandler.getEnergy() > AstromineConfig.get().gravityGauntletConsumed && otherHandler.getEnergy() > AstromineConfig.get().gravityGauntletConsumed) {
+				selfHandler.extract(AstromineConfig.get().gravityGauntletConsumed);
+				otherHandler.extract(AstromineConfig.get().gravityGauntletConsumed);
+				stack.getOrCreateTag().putBoolean("Charged", true);
+				offStack.getOrCreateTag().putBoolean("Charged", true);
+				return stack;
+			}
+		}
+		return super.finishUsing(stack, world, user);
+	}
+
+	@Override
+	public UseAction getUseAction(ItemStack stack) {
+		return UseAction.BLOCK;
+	}
+
+	@Override
+	public int getMaxUseTime(ItemStack stack) {
+		return 30;
+	}
+
+	@Override
+	public boolean postHit(ItemStack stack, LivingEntity target, LivingEntity attacker) {
+		if (attacker.world.isClient)
+			return super.postHit(stack, target, attacker);
+		ItemStack offStack = attacker.getStackInHand(Hand.OFF_HAND);
+		if (offStack.getItem() == AstromineItems.GRAVITY_GAUNTLET) {
+			if (stack.getOrCreateTag().getBoolean("Charged") && offStack.getOrCreateTag().getBoolean("Charged")) {
+				target.takeKnockback(1, attacker.getX() - target.getX(), attacker.getZ() - target.getZ());
+				target.addVelocity(0f, 0.5f, 0f);
+				stack.getOrCreateTag().putBoolean("Charged", false);
+				offStack.getOrCreateTag().putBoolean("Charged", false);
+				return true;
+			}
+		}
+		return super.postHit(stack, target, attacker);
+	}
+
+	@Override
+	public boolean hasGlint(ItemStack stack) {
+		return stack.getOrCreateTag().getBoolean("Charged");
+	}
+
+	// TODO: dynamic once not broken
+	@Override
+	public Multimap<EntityAttribute, EntityAttributeModifier> getAttributeModifiers(EquipmentSlot slot) {
+		Multimap<EntityAttribute, EntityAttributeModifier> eams = HashMultimap.create();
+		if (slot == EquipmentSlot.MAINHAND) {
+			eams.put(EntityAttributes.GENERIC_ATTACK_DAMAGE, new EntityAttributeModifier(ATTACK_DAMAGE_MODIFIER_ID, "attack", 4f, EntityAttributeModifier.Operation.ADDITION));
+		}
+		return super.getAttributeModifiers(slot);
+	}
+
+	@Environment(EnvType.CLIENT)
+	@Override
+	public void appendTooltip(ItemStack stack, World world, List<Text> tooltip, TooltipContext context) {
+		EnergyHandler energyHandler = Energy.of(stack);
+		tooltip.add(EnergyUtilities.compoundDisplayColored(energyHandler.getEnergy(), energyHandler.getMaxStored()));
```

Comment:
call super.appendTooltip

Reply:
that would add the crafting use only warning

Reply:
https://github.com/Chainmail-Studios/Astromine/blob/development/src/main/java/com/github/chainmailstudios/astromine/common/item/base/EnergyVolumeItem.java#L70